skglm/solvers/cd_solver.py

```diff
@@ -145,15 +142,17 @@ def cd_solver_path(X, y, datafit, penalty, alphas=None,
                 Xw = np.zeros(X.shape[0], dtype=X.dtype)
 
         sol = cd_solver(
-            X, y, datafit, penalty, w, Xw,
+            X, y, datafit, penalty, w, Xw, fit_intercept=fit_intercept,
             max_iter=max_iter, max_epochs=max_epochs, p0=p0, tol=tol,
             use_acc=use_acc, verbose=verbose, ws_strategy=ws_strategy)
 
         coefs[:, t] = w
+        if fit_intercept:
+            intercepts[0, t] = sol[1]
         stop_crits[t] = sol[-1]
 
         if return_n_iter:
-            n_iters[t] = len(sol[1])
+            n_iters[t] = len(sol[2])
```

Comment:
this means the signature of cd_solver is changed. It can break existing code, I would avoid this

Reply:
then we have stop_crits[t] = sol[-1] so I cannot add the intercept at the end either. 
Should i do return coefs, n_iters, intercept stop_crits? 
Does not make lots of sense either to me

Reply:
can you return a coefs which is longer by 1 if `fit_intercept` ? This way the signature does not change. And we split in the output of the solver